packages/freezed/lib/src/templates/concrete_template.dart

```diff
@@ -323,7 +323,7 @@ ${whenOrNullPrototype(allConstructors)} {
     return '''
 @override
 String toString($parameters) {
-  return '${constructor.fullName}(${properties.join(', ')})';
```

Comment:
Rather than dealing with escaping the name, a simpler solution would be to use:

```dart
return r'${constructor.fullName}...';
```

This way this will generate a string that doesn't interact with $